### DISCUSSION TITLE
关于thrdpool模块的讨论

### DISCUSSION THREAD

**Discussion by holmes1412** [sogou/workflow#886] [Ideas]
最近深入阅读了thrdpool模块的源码，感觉这个模块在设计上有许多新颖的点，单独拿出来使用也非常方便。后来发了篇笔记，引发了许多热心开发者的讨论~~~

想要知道大家对线程池都有怎样的需求和想法呢？包括接口和架构设计的点子都非常欢迎。

源码链接：
https://github.com/sogou/workflow/blob/master/src/kernel/thrdpool.c
https://github.com/sogou/workflow/blob/master/src/kernel/thrdpool.h

**Comment by real-hong** [comment]
关于thrdpool_t中的cond，请问这个的作用是什么呢？

**Reply by Barenboim** [reply]
> 关于thrdpool_t中的cond，请问这个的作用是什么呢？

这个已经改了。用msgqueue重写thrdpool时，忘了把这个cond删掉。

**Comment by sourcema** [comment]
请问下msgqueue_create(0, 0)为什么消息的个数初始化为0？

**Reply by Barenboim** [reply]
这里0表示无限。因为最大0个消息，好像没有意义，所以就让0来表示无限制了。

**Reply by sourcema** [reply]
msg_max是size_t，了解了
